cpp/Platform.Interfaces.Tests/Platform.Interfaces.Tests.cpp

```diff
@@ -3,162 +3,162 @@
     {
-        struct EmptyProperties
+        struct EmptyProperties: public IProperties<int, int, int>
         {
-            void SetValue(std::string object, int argument, std::byte value) {}
+            void SetValue(int, int, int) {}
 
-            std::byte GetValue(const std::string& object, int argument) { return std::byte{}; }
+            int GetValue(int, int) { return {}; }
         };
-        static_assert(IProperties<EmptyProperties, std::string, int, std::byte>);
+        static_assert(CProperties<EmptyProperties, int, int, int>);
 
         {
-            IProperties<std::string, int, std::byte> auto properties = EmptyProperties{};
+            CProperties<int, int, int> auto properties = EmptyProperties{};
 
-            ASSERT_TRUE((IProperties<EmptyProperties, std::string, int, std::byte>));
-            ASSERT_TRUE((IProperties<EmptyProperties, std::string&, int, std::byte>));
-            ASSERT_TRUE((IProperties<EmptyProperties, const std::string&, int, std::byte>));
+            ASSERT_TRUE((CProperties<EmptyProperties, int, int, int>));
+            ASSERT_TRUE((CProperties<EmptyProperties, int&, int, int>));
+            ASSERT_TRUE((CProperties<EmptyProperties, const int&, int, int>));
         }
     }
 
-    TEST(CompileTests, IProvider)
+    TEST(CompileTests, Provider)
     {
-        struct EmptyProvider
+        struct EmptyProvider: public IProvider<int>
         {
-            std::byte Get() { return std::byte{}; }
+            int Get() { return {}; }
         };
-        static_assert(IProvider<EmptyProvider, std::byte>);
+        static_assert(CProvider<EmptyProvider, int>);
 
-        struct EmptyIntProvider
+        struct EmptyIntProvider: public IProvider<int, int>
         {
-            std::byte Get(int argument) { return std::byte{}; }
+            int Get(int argument) { return {}; }
         };
-        static_assert(IProvider<EmptyIntProvider, std::byte, int>);
+        static_assert(CProvider<EmptyIntProvider, int, int>);
 
         {
-            IProvider<std::byte> auto provider = EmptyProvider{};
+            CProvider<int> auto provider = EmptyProvider{};
 
-            ASSERT_TRUE((IProvider<EmptyProvider, std::byte>));
+            ASSERT_TRUE((CProvider<EmptyProvider, int>));
         }
 
         {
-            IProvider<std::byte, int> auto provider = EmptyIntProvider{};
+            CProvider<int, int> auto provider = EmptyIntProvider{};
 
-            ASSERT_TRUE((IProvider<EmptyIntProvider, std::byte, int>));
-            ASSERT_TRUE((IProvider<EmptyIntProvider, std::byte, float>));
+            ASSERT_TRUE((CProvider<EmptyIntProvider, int, int>));
+            ASSERT_TRUE((CProvider<EmptyIntProvider, int, float>));
         }
     }
 
-    TEST(CompileTests, ISetter)
+    TEST(CompileTests, Setter)
     {
-        struct EmptySetter
+        struct EmptySetter: public ISetter<int>
         {
-            void Set(std::byte value) { }
+            void Set(int value) { }
```

Comment:
Redundant argument name
```suggestion
            void Set(int) { }
```